src/containers/Moderation/moderationHooks.js

```diff
@@ -0,0 +1,195 @@
+
+    const handleLogin = e => {
+        e.preventDefault();
+
+        async function postLogin() {
+            try {
+                const serverData = await Api.postLogin(loginData);
+                sessionStorage.moderatorToken = serverData.token;
+                dispatch({
+                    type: SET_LOGGED_IN,
+                    payload: true
+                });
+            } catch (e) {
+                window.alert(e);
+            }
+        }
+        postLogin();
+    };
+
+    return {
+        handleLogin
+    };
+};
+
+export const useModerationStories = () => {
+    const { moderationState } = useModerationContext();
+
+    const [stories, setStories] = useState();
+    useEffect(() => {
+        if (moderationState.loggedIn) {
+            (async () => {
+                await setStories(
+                    await Api.getModerationStories('createdAt', 'ASC')
+                );
+            })();
+        }
+    }, [moderationState.loggedIn]);
+    return {
+        storiesToModerate: stories?.result
+    };
+};
+
+export const useModerationStory = story => {
+    const { moderationState, dispatch } = useModerationContext();
+    useEffect(() => {
+        if (story._id !== moderationState._id) {
+            const processedStory = extractFieldsFromObj(story, [
+                '_id',
+                'additionalnfo',
+                'background',
+                'mail',
+                'howDidYouManged',
+                'name',
+                'quote',
+                'storyContent',
+                'whatHelpedYou',
+                'whatTriggeredChange',
+                'contact'
+            ]);
+            dispatch({
+                type: SET_MODERATE_STORY_DATA,
+                payload: { ...moderationState, ...processedStory }
+            });
+            dispatch({
+                type: SET_TAGS,
+                payload: []
+            });
+        }
+    }, []);
+    return {};
+};
+
+export const useModerateStorySubmit = () => {
+    const { moderationState, dispatch } = useModerationContext();
+    const [submitted, setSubmitted] = useState(false);
+    let moderationDataToPost = { ...moderationState };
+    delete moderationDataToPost.loggedIn;
+    moderationDataToPost.originalStory = moderationState._id;
+    delete moderationDataToPost._id;
+    moderationDataToPost.tags = getArrayOfTagsIds(moderationDataToPost.tags);
+
+    const handleSubmit = e => {
+        e.preventDefault();
+
+        async function postData() {
+            try {
+                await Api.postModerateStory(moderationDataToPost);
+                dispatch({
+                    type: SET_MODERATE_STORY_DATA,
+                    payload: {
+                        ...moderationState,
+                        ...NEW_MODERATE_STORY_INIT_DATA
+                    }
+                });
+                setSubmitted(true);
+            } catch (e) {
+                window.alert(e);
```

Comment:
window.alert is too dramatic. Use console.error until we will handle errors globally